src/Chapter4.hs

```diff
@@ -489,6 +507,26 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Semigroup (List a) where
+    (<>) :: List a -> List a -> List a
+    (<>) Empty Empty = Empty
+    (<>) Empty (Cons y ys) = Cons y ys
+    (<>) (Cons x xs) Empty = Cons x xs
+    (<>) (Cons x xs) (Cons y ys) = (Cons x (mappend xs (Cons y ys)))
```

Comment:
You can reduce the number of patterns to three here. You can see that in case of `Empty`, you always return the other argument. So you can simplify the implementation a bit :slightly_smiling_face:

Reply:
Pattern matching kind of takes getting used to, at least for me. Thanks for spotting that.